solr/core/src/java/org/apache/solr/handler/admin/api/CoreReplicationAPI.java

```diff
@@ -88,4 +90,137 @@ public IndexVersionResponse(Long indexVersion, Long generation, String status) {
       this.status = status;
     }
   }
+
+  /** Response for {@link CoreReplicationAPI#fetchFileList(long)}. */
+  public static class FileListResponse extends SolrJerseyResponse {
+    @JsonProperty("filelist")
+    public List<FileMetaData> fileList;
+
+    @JsonProperty("confFiles")
+    public List<FileMetaData> confFiles;
+
+    @JsonProperty("status")
+    public String status;
+
+    @JsonProperty("message")
+    public String message;
+
+    @JsonProperty("exception")
+    public Exception exception;
+
+    public FileListResponse() {}
+
+    public void addToFileList(List<FileMetaData> fileMetaData) {
+      if (fileList == null) {
+        fileList = new ArrayList<>();
+      }
+      fileList.addAll(fileMetaData);
+    }
+
+    public void addToConfFiles(List<FileMetaData> confFilesMetaData) {
+      if (confFiles == null) {
+        confFiles = new ArrayList<>();
+      }
+      confFiles.addAll(confFilesMetaData);
+    }
+
+    public void setFileList(List<FileMetaData> fileList) {
```

Comment:
[Q] Do setters/getters here provide any value?  We kindof have to make the fields in this class 'public' bc of how we use Jackson, and it doesn't look like the getters/setters are doing anything that'd be different from users just accessing the fields directly.

Reply:
I took another look and removed them. After some though, I don't think they do provide any value. These are supposed to be POJO objects and should be nothing but a response object without being overly complex.